### PR TITLE
Align servers' evaluation approach with debugger core

### PR DESCRIPTION
**Pull request by st0012**
`frame_eval_core` enabled nested TracePoint callbacks, so I think we should use that in our server implementations too.

### CODE REVIEW

File: lib/debug/server_dap.rb [hunk 1]
```diff
@@ -699,16 +699,15 @@ def process_dap args
         frame = get_frame(fid)
         message = nil
 
-        if frame && (b = frame.binding)
-          b = b.dup
+        if frame && (b = frame.eval_binding)
           special_local_variables current_frame do |name, var|
```
Comment (st0012) [inline lib/debug/server_dap.rb:703]: @ko1 perhaps this should be `frame` instead?

Comment (ko1) [reply lib/debug/server_dap.rb:703]: it should be.

**Comment by ko1** [comment]
I need to try it on vscode.

**Comment by st0012** [comment]
Hmm in my tests it doesn't seem to enable nested breakpoints yet.

**Comment by st0012** [comment]
We may need more planning to get servers support nested breakpoints (it could also be impossible). So closing this for now.

**Pull request closed without merging**